includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1353,32 +1401,7 @@ public function setup_intent( $order, $prepared_source ) {
 	 */
 	public function create_and_confirm_intent_for_off_session( $order, $prepared_source, $amount = NULL ) {
 		// The request for a charge contains metadata for the intent.
-		$full_request = $this->generate_payment_request( $order, $prepared_source );
-
-		$request = array(
-			'amount'               => $amount ? WC_Stripe_Helper::get_stripe_amount( $amount, $full_request['currency'] ) : $full_request['amount'],
-			'currency'             => $full_request['currency'],
-			'description'          => $full_request['description'],
-			'metadata'             => $full_request['metadata'],
-			'payment_method_types' => array(
-				'card',
-			),
-			'off_session'          => 'true',
-			'confirm'              => 'true',
-			'confirmation_method'  => 'automatic',
-		);
-
-		if ( isset( $full_request['statement_descriptor'] ) ) {
-			$request['statement_descriptor'] = $full_request['statement_descriptor'];
-		}
-
-		if ( isset( $full_request['customer'] ) ) {
-			$request['customer'] = $full_request['customer'];
-		}
-
-		if ( isset( $full_request['source'] ) ) {
-			$request['source'] = $full_request['source'];
-		}
+		$request = $this->generate_create_intent_request_for_off_session( $order, $prepared_source, $amount );
```

Comment:
I don't really see the value in extracting this logic into it's own function at this time. I think we're ok to just do this instead:

```suggestion
		$request = apply_filters( 'wc_stripe_generate_create_intent_request', $request, $order, $prepared_source )
```

Reply:
I only added this for consistency with how it is done in other methods such like `generate_create_intent_request`.

Reply:
I appreciate the thought you put into that, but I think it will be better to apply the filter directly instead of extracting the first half of `create_and_confirm_intent_for_off_session` into its own function.

I think extracting parts of this function into separate functions would be better served as part of a bigger refactor where we address more than just this single instance of code that would be better served as its own function 🙂

Reply:
@reykjalin this too has been updated.